packages/libs/react-ui/src/components/Modal/Modal.tsx

```diff
@@ -1,68 +1,63 @@
 'use client';
-import { Card } from '@components/Card';
-import { SystemIcon } from '@components/Icon';
-import { Heading } from '@components/Typography/Heading/Heading';
-import FocusTrap from 'focus-trap-react';
-import type { FC } from 'react';
-import React from 'react';
-import {
-  background,
-  closeButton,
-  modal,
-  titleWrapper,
-  wrapper,
-} from './Modal.css';
-import { useModal } from './useModal';
 
-export interface IModalProps {
-  children: React.ReactNode;
-  title?: string;
-  onClose?: () => void;
+import { mergeRefs } from '@react-aria/utils';
+import clsx from 'classnames';
+import type { FC, ReactElement, Ref } from 'react';
+import React, { cloneElement, useRef } from 'react';
+import type { AriaModalOverlayProps, ModalOverlayAria } from 'react-aria';
+import { Overlay, useModalOverlay } from 'react-aria';
+import type { OverlayTriggerState } from 'react-stately';
+import { underlayClass } from './Modal.css';
+
+export interface IModalProps extends AriaModalOverlayProps {
+  className?: string;
+  state: OverlayTriggerState;
+  children:
+    | ReactElement
+    | ((
+        modalProps: ModalOverlayAria['modalProps'],
+        ref: Ref<HTMLDivElement>,
+      ) => ReactElement);
 }
 
-export const Modal: FC<IModalProps> = ({ children, title, onClose }) => {
-  const { clearModal } = useModal();
+export const Modal: FC<IModalProps> = ({
+  className,
+  children,
+  state,
+  isDismissable = true,
+  isKeyboardDismissDisabled,
+}) => {
+  const nodeRef = useRef<HTMLDivElement | null>(null);
+  const underlayRef = useRef<HTMLDivElement | null>(null);
```

Comment:
Is this underlayRef being used for anything? I only see it being applied to the underlay div